coding/CMakeLists.txt

```diff
@@ -0,0 +1,81 @@
+include_directories(${PROJECT_SOURCE_DIR}/3party/tomcrypt/src/headers)
+
+set(
+    SRC
+    base64.cpp
+    base64.hpp
+    bit_streams.hpp
+    buffer_reader.hpp
+    byte_stream.hpp
+    coder.hpp
+    coder_util.hpp
+    compressed_bit_vector.cpp
+    compressed_bit_vector.hpp
+    constants.hpp
+    dd_vector.hpp
+    diff.hpp
+    diff_patch_common.hpp
+    elias_coder.hpp
+    endianness.hpp
+    file_container.cpp
+    file_container.hpp
+    file_name_utils.cpp
+    file_name_utils.hpp
+    file_reader.cpp
+    file_reader.hpp
+    file_sort.hpp
+    file_writer.cpp
+    file_writer.hpp
+    fixed_bits_ddvector.hpp
+    hex.cpp
+    hex.hpp
+    huffman.cpp
+    huffman.hpp
+    matrix_traversal.hpp
+    memory_region.hpp
+    mmap_reader.cpp
+    mmap_reader.hpp
+    multilang_utf8_string.cpp
+    multilang_utf8_string.hpp
+    parse_xml.hpp
+    png_memory_encoder.cpp
+    png_memory_encoder.hpp
+    polymorph_reader.hpp
+    read_write_utils.hpp
+    reader.cpp
+    reader.hpp
+    reader_cache.hpp
+    reader_streambuf.cpp
+    reader_streambuf.hpp
+    reader_wrapper.hpp
+    reader_writer_ops.cpp
+    reader_writer_ops.hpp
+    sha2.cpp
+    sha2.hpp
+    simple_dense_coding.cpp
+    simple_dense_coding.hpp
+    streams.hpp
+    streams_common.hpp
+    streams_sink.hpp
+    succinct_mapper.hpp
+    uri.cpp
+    uri.hpp
+    url_encode.hpp
+    value_opt_string.hpp
+    var_record_reader.hpp
+    var_serial_vector.hpp
+    varint.hpp
+    varint_misc.hpp
+#    varint_vector.cpp
+#    varint_vector.hpp
+    write_to_sink.hpp
+    writer.hpp
+    zip_creator.cpp
+    zip_creator.hpp
+    zip_reader.cpp
+    zip_reader.hpp
+)
+
+omim_add_library(coding ${SRC})
+
+#add_subdirectory(coding_tests)
```

Comment:
Because the test depends on a lot of other libraries in the project, we need to make those libraries compile first. Then we will be able to proceed to the tests.